components/Map/index.js

```diff
@@ -0,0 +1,117 @@
+import Link from "next/link";
+
+import styled from "styled-components";
+
+import { MapContainer, Marker, Popup, TileLayer } from "react-leaflet";
+import Lottie from "lottie-react";
+import * as L from "leaflet";
+import "leaflet/dist/leaflet.css";
+import "leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.css";
+import "leaflet-defaulticon-compatibility";
+
+import LoadingAnimation from "@/public/lottiefiles/LoadingAnimation.json";
+
+import CurrentMarker from "@/components/Map/CurrentMarker";
+
+const StyledMapAndIconContainer = styled.div`
+  position: relative;
+`;
+const StyledMapContainer = styled(MapContainer)`
+  height: 30vh;
+  width: calc(100%+30px);
+  margin: 0 -15px;
+  z-index: 0;
+`;
+
+const StyledLink = styled(Link)`
+  h2 {
+    color: var(--primaryDarkColor);
+    &:hover {
+      cursor: pointer;
+      color: var(--accentColor);
+    }
+  }
+  text-decoration: none;
+`;
+
+const StyledLottie = styled(Lottie)`
+  width: 30%;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  z-index: 2;
```

Comment:
using z-index is often caused by mistakes made with the stacking context

Reply:
thank, you're right that it's not very clean to use the z-index here but the leaflet map (or rather the MapContainer) is set by leaflet to not be part of the `<main>` . And we didn't find any way to change that so that it's part of the `<main>`. If it would be part of the `<main>`, then the stacking order is enough, of course. But if it's a separate Element of the DOM, the map overlaps the `<header>`and `<main>`, what we prevent here by using the z-index. For now, that should be an ok solution for our App...